[WIP] Test Bed - Testing possible future improvements

onaforeignshore opened the pull request:
Here are the updates to #863 to incorporate the new `registerSegment`. This supercedes #833 as all the ideas have been incorporated here.

It contains all the code needed to make use of the new format, as well as a new function to rename variables to match the new structure - `printDeprecationVarWarning`

TODO: Decide where custom segments will be located and load them at initialization.

onaforeignshore committed:
Update to command_execution_time test

onaforeignshore committed:
Updated Travis tests for segments

onaforeignshore committed:
Updates to Travis tests

onaforeignshore committed:
Update for vcs test

onaforeignshore committed:
Updated segment_in_use

Created single list of all segments - POWERLEVEL9K_PROMPT_SEGMENTS

onaforeignshore committed:
Updated comments for segments

onaforeignshore committed:
Updated powerlevel9k.spec

onaforeignshore committed:
2nd update to powerlevel9k.spec

onaforeignshore committed:
Updated dir.spec

onaforeignshore committed:
2nd update to dir.spec

onaforeignshore committed:
dir.spc - updated setUp()

onaforeignshore committed:
Update to dir segment #776

onaforeignshore committed:
Added function: serialize_segment

This is the beginning of moving from the current generator (sync) to a
new generator (async).

TODO: Update all segments to use serialize_segment instead of calling
left_prompt_segment or right_prompt_segment directly.

onaforeignshore committed:
Added generators folder and updated segments

* Moved all generator code to a separate folder - generator/default.p9k
* Changed all segment calls to use serialize_segment
* Added zsh-async and generator/zsh-async.p9k for EXPERIMENTAL async
support

onaforeignshore committed:
Fixed double visual identifier code error

onaforeignshore committed:
Icon drawing correction

onaforeignshore committed:
Reverted #776

Seems that removing the -P from the print command does make a
difference.

onaforeignshore committed:
Code cleanup

onaforeignshore committed:
Rewrote vi_mode to support sync & async

Combined synchronous and asynchronous code into a single prompt segment

onaforeignshore committed:
vi_mode: Supports term ID and cursor shape change

To enable cursor shape change, set:
POWERLEVEL9K_CURSOR_SHAPE=true

On supported terminals, cursor shape can be box [], hbar _ or vbar |

Example:
POWERLEVEL9K_CURSOR_SHAPE_NORMAL="box"
POWERLEVEL9K_CURSOR_SHAPE_INSERT="vbar"
POWERLEVEL9K_CURSOR_SHAPE_VISUAL="box"
POWERLEVEL9K_CURSOR_SHAPE_DEFAULT="hbar"

onaforeignshore committed:
Added java_version, comments & API

- Added prompt_java_version & updated README.md
- Updated comments for ALL functions for use with sh2md.
- Added API folder. This contains all the Markdown documentation for all
of the *.p9k files.

onaforeignshore committed:
Merge branch 'next' into code_separation

onaforeignshore committed:
MAJOR UPDATE

CHANGES:
- flattened `segments` folder, but kept `segments\custom`.
- added function `updateEnvironmentVars` to automatically change user variables. NOTE: There may be minor issues with arrays, so this still needs review.
- added local keyword to variables where appropriate.
- streamlined P9K_PROMPT_ELEMENT generation.
- added ability to autoload segments to `build_x_prompt` when user changed `P9K_x_PROMPT_ELEMENT` arrays.
- added tracking of loaded segments using `loadedSegments`.
- fixed coding error in `prompt_status`.
- change all internal functions to use camelCase. Exceptions are listed below:
  - prompt functions still use `prompt_x` so that we can differentiate between internal (core) functionality and segments.
  - `prompt_powerlevel9k_setup`, `prompt_powerlevel9k_teardown`, and `p9k_preexec`.
- replaced variables `POWERLEVEL9K_` -> `P9K_` and `powerlevel9k_` -> `p9k_`.

TODO:
- Decide on a default variable naming scheme.
- Update code & comments in `segments/vcs.p9k` and `functions/vcs.zsh`.

onaforeignshore committed:
Minor update

Forgot to prefix variables in tests with $
*hides in the corner*

onaforeignshore committed:
Update for Travis

Travis requires [[ $var == true ]] to work and not just [[ $var ]]

onaforeignshore committed:
Travis update #2

Travis seems to have a problem with this line:

    typeset -gU P9K_PROMPT_ELEMENTS=("${P9K_LEFT_PROMPT_ELEMENTS[@]}" "${P9K_RIGHT_PROMPT_ELEMENTS[@]}")

replaced it with:

    typeset -gU P9K_PROMPT_ELEMENTS
    P9K_PROMPT_ELEMENTS=("${P9K_LEFT_PROMPT_ELEMENTS[@]}" "${P9K_RIGHT_PROMPT_ELEMENTS[@]}")

onaforeignshore committed:
Segment Auto Loading updates

Custom segments:
- Check if the custom segment is a variable/definition. If so, use it.
- Next, check if a custom segment file exists. If so, use it.

Segments:
- Check if a segment file exists. If so, use it.

onaforeignshore committed:
Updated tests

Since we now have autoloading of segments, removed sourcing of segments
in the setUp function of the tests.

onaforeignshore committed:
Updates to zsh-async

Coding style updates, and sync with default generator

onaforeignshore committed:
Updates to zsh-async.md

onaforeignshore committed:
zsh-async fixes

- segment separator colors
- first right prompt segment background color
- vi-mode coding error
- iTerm integration
- made some variables local
- changed MULTILINE_SECOND_PROMPT_PREFIX to MULTILINE_LAST_PROMPT_PREFIX

It works better now, but some of the segments still have occasional
hiccups and print the wrong thing.

onaforeignshore committed:
default generator updates

- Added support for bolded segments
- Added support to have the right segment icon displayed on the left

onaforeignshore committed:
Updated README.md

- Added missing links in index
- Reordered everything to match index

dritter committed:
Fix error where Darwin is not recognized

This was due to an incomplete rename of "OSX" to "macOS".

onaforeignshore committed:
Merge pull request #2 from dritter/oafs_fix_osx

Fix mac OS not recognized

onaforeignshore committed:
Minor README.md update

onaforeignshore committed:
Updated CHANGELOG.md to reflect changes

Used version 0.7.0 for the version number as there are numerous major
changes in this release.

onaforeignshore committed:
Updated misspelled name for Dominik Ritter

onaforeignshore committed:
Merge branch 'next' into code_separation

onaforeignshore committed:
Merged - Fix wrapper slowness in vcs segment #797

onaforeignshore committed:
Merged - Add vcs branch name truncating. #798

onaforeignshore committed:
Updated test/vcs.spec to use buildLeftPrompt

onaforeignshore committed:
README.md - add rvm #799

onaforeignshore committed:
Updated functions/vcs.zsh - setDefault

onaforeignshore committed:
functions/vcs.zsh - fixed P9K_ environment vars.

onaforeignshore committed:
vcs.spec - Fixed P9K_ environment vars

onaforeignshore committed:
Fixed typo in README.md - p8k_

onaforeignshore committed:
Various changes. See description

functions/icons.zsh
Updated registerIcon to check for icon overrides and use those if found
Updated printIcon to now only print the icon
Removed get_icon_names
Added showDefinedIcons - lists all defined icons alphabetically

default.p9k
Changed $3PromptSegment to ${ALIGNMENT}PromptSegment
Renamed serializeSegment to p9kPrepareSegment

zsh-asynv.p9k
Removed unneccesary testing in leftPromptSegment and rightPromptSegment
Fixed "left" in updateRightPrompt
Fixed an issue where invisible segments were adding a space to a prompt
segment
Renamed serializeSegment to p9kPrepareSegment
Commented out the auto-loading of segments in buildLeftPrompt and
buildRightPrompt

history.p9k
Fixed incorrect icon name for segment

vcs.p9k
Added loads of comments in preparation for streamlining this segment

All segments
Renamed serializeSegment to p9kPrepareSegment

onaforeignshore committed:
Removed obsolete line from powerlevel9k.spec

onaforeignshore committed:
Updated incon functions, See details below

functions/icons.zsh
- changed core icon definitions to use registerIcon()

segments/vcs.p9k
- moved icon override for P9K_HIDE_BRANCH_ICON to this file instead of
in icons.zsh

powerlevel9k.zsh-theme
- changed the sourcing order of icons and utilities so that defined() is
available to registerIcon()

dritter committed:
Merge remote-tracking branch 'onaforeignshore/code_separation' into staging_700

Merge of #771 into #859

dritter committed:
Fix exception in status segment

dritter committed:
Fix variable names after merge

dritter committed:
Fix path to test_stats segment

dritter committed:
Remove testwise enabling of mercurial queues

dritter committed:
Fix rust_version segment

dritter committed:
Fix tests

dritter committed:
Add forgotten Fix for RPROMPT placement from #847 after merge

dritter committed:
Add forgotten Fix for RPROMPT placement from #847 after merge

For zsh-async generator

onaforeignshore committed:
See comments below

- renamed array icons to p9k_icons
- changed registerIcon comment description
- added registerSegment

onaforeignshore committed:
Removed space between upSearch and ()

onaforeignshore committed:
Added segmentShouldBePrinted

This helper function tests the segment condition to determine whether it should
be printed or not.

onaforeignshore committed:
See notes below

Multiple changes to make use of registerSegment and to use precompiled
data instead of compiling it each time.
- left|rightPromptSegment
  - make use of STATEFUL_NAME to determine colors, icons and icon colors
  - fixed segment should be joined logic (all segments now have a
  background color)
  - fixed all printIcon calls to make use of the p9k_icons array instead
  - icon whitespace is now determined here instead of in
  p9kPrepareSegment. This allows us to test for empty icons and handle
  that.
- leftPromptEnd
  - uses p9k_icons instead of printIcon
- p9kPrepareSegment
  - makes use of segmentShouldBePrinted to determine the condition and
  only calls left|rightPromptSegment if required
  - determines if an icon override has been used and sets the correct
  icon
- removed background and foreground parameters from
left|rightPromptSegment and prompt_custom as these are now set in
registerSegment

onaforeignshore committed:
See notes below

- moved printDeprecationWarning to powerlevel9k.zsh-theme

onaforeignshore committed:
See notes below

- moved printDeprecationWarning here from generator/default.p9k
- added printDeprecationVarWarning

onaforeignshore committed:
See notes below

- added function updateVarName
- added function printDeprecationVarWarning

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color
- moved setDefault outside of function
- added '/private/etc' for macOS compatibility

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed random colors
- changed icons to p9k_icons
- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- added notes

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
Fixed comment icon spacing

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color
- moved setDefault outside of function
- added macOS test for gnu stat

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color
- updated P9K_RBENV_ALWAYS logic

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- removed icon from build_test_stats

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- removed icon from build_test_stats

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- added states
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- removed state arrays
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore committed:
See notes below

- removed authors from file
- replaced registerIcon with registerSegment
- updated p9kPreparePrompt call to not use color

onaforeignshore commented:
After working all day on this, the next step is to update the tests

onaforeignshore committed:
VI_MODE - See notes below

- reinstated the P9K_VI_mode_MODE_STRING variables
- added segment condition to display the segment if the string is empty,
but an icon is defined.
- you can now set a custom string, as well as a custom icon, or both
for each state.

onaforeignshore committed:
Fix missed icons to p9k_icons renaming (179-188)

onaforeignshore committed:
Fixed test/powerlevel9k.spec

- Removed extra background/foreground setting in joined segments. If the
background color is the same as the previous segment, why set it again?

onaforeignshore committed:
Renamed array "icons" to "p9k_icons"

onaforeignshore committed:
See notes below

- added registerSegment call for custom segment
- replaced %} with }

onaforeignshore committed:
Changed variable names to new ones

onaforeignshore committed:
See notes below

- replaced %} with }
- added registerSegment for custom world segment

onaforeignshore committed:
replaced %} with }

onaforeignshore committed:
See notes below

- replaced %} with }
- renamed magenta to 001
- changed variable P9K_LARAVEL_ICON to P9K_LARAVEL_VERSION_ICON
- added registerSegment "WORLD" for custom segment

onaforeignshore committed:
forgot to rename unset variables

onaforeignshore committed:
See notes below

- renamed GO_ICON to GO_VERSION_ICON
- added call to functions/icons.zsh after icon change
- added registerSegment "WORLD" for custom segments

onaforeignshore committed:
Merge branch 'next' into 070-registerSegment

bhilburn commented:
Just as a note: right now, this branch / PR looks really complicated, but that's because it's built on top of #863, which is not yet in `next` - so it's duplicating that commit tree in the PR. Once we have merged #863, this branch / PR should be rebased onto `next`, which should make it much easier to manage.

onaforeignshore committed:
Dir - New Strategy - see notes below

functions/utilities.zsh
- added "left" to truncatePath and fixed description
- moved all local declarations out of loops

segments/dir.p9k
- added "truncate_from_left" strategy

onaforeignshore committed:
Streamlined segmentInUse

onaforeignshore committed:
Removed erroneous line left from moving local decl

onaforeignshore committed:
Segment loading updates - see notes below

- removed loadedSegments array - remnant from autoloading which was
removed
- added loading custom segments from ~/.config/powerlevel9k/segments

onaforeignshore committed:
Set default for P9K_DIR_PATH_ABSOLUTE to false

onaforeignshore committed:
Added test to check custom folder exists

onaforeignshore committed:
Replaced $'\U' with $'\u' for consistency

onaforeignshore committed:
Changed printIcon to p9k_icons in p9kPreparePrompts

onaforeignshore committed:
Updated tests

onaforeignshore committed:
Dir segment fixes - see notes below

functions/utilities.zsh
- added locals delim and delim_len for code clarity
- fixed minimum length to 3

segments/dir.p9k
- added "" around $current_path
- added "" around $P9K_SHORTEN_DELIMITER for using an empty delimiter

onaforeignshore committed:
Updated tests - see notes below

test/powerlevel9k.spec
- fixed testNewlineOnRpromptCanBeDisabled to use literals instead of
unicode

test/segments/go_version.spec
- fixed colors to {grey93}

onaforeignshore committed:
Tests - see notes below

- changed all variables to local and removed unset

onaforeignshore committed:
Reverted "fix" for minimum dir length

onaforeignshore committed:
Various fixes - see notes below

- added test to ensure path is longer than one character before applying
strategies
- truncate_to_unique
  - added code to determine unique folder location of home folder
  - changed paths array to use absolute path
  - added "cheat" to replace unique home folder with ~

onaforeignshore committed:
Removed segment autoloading code

onaforeignshore committed:
Various changes - see notes below

- since segment autoloading has been removed, updated all tests to
source the appropriate segment files
- segment files have to be resourced after changing color or icon, as
these are static

onaforeignshore committed:
Fixed local declarations

onaforeignshore committed:
Attempt to fix error in test ln 31: bad pattern

onaforeignshore committed:
Changed getting unique home path to function

onaforeignshore committed:
Added sourcing background_jobs & root_indicator

onaforeignshore committed:
Attempt to fix "number expected"

onaforeignshore committed:
Forgot $ before var in test

onaforeignshore committed:
Attempt 2 to fix dir

onaforeignshore committed:
Attempt 3 to fix dir

onaforeignshore committed:
Attempt 4 to fix dir

onaforeignshore committed:
Attempt 5 to fix dir

onaforeignshore committed:
Made variables local

onaforeignshore committed:
Updated tests - see notes below

Changed local declaration of P9K_LEFT_PROMPT_ELEMENTS from

local P9K_LEFT_PROMPT_ELEMENTS=()

to

local -a P9K_LEFT_PROMPT_ELEMENTS
P9K_LEFT_PROMPT_ELEMENTS=()

This is done to support ZSH 5.0.2, which cannot do one line array declations

onaforeignshore committed:
See notes below

Added P9K_DIR_PATH_ABSOLUTE=true to testTruncateToUniqueWorks

onaforeignshore committed:
Changed local paths declaration to local -a paths

This is for compatibility with ZSH 5.0.2

onaforeignshore committed:
Updated local mathching to local -a matching

Another update for 5.0.2 compatibility

onaforeignshore committed:
Updated truncate_to_unique strategy

onaforeignshore committed:
Code cleanup - See notes below

- left|rightPromptSegment
  - combined NAME and STATE into STATEFUL_NAME
  - cleaned up BOLD code
  - added local content=${5} for code clarity and easy parameter
  changing
- p9kPrepareSegment
  - added local declaration for CONDITION
  - combined NAME and STATE into STATEFUL_NAME

onaforeignshore committed:
See notes below

- used the code from dritter/oafs_fix_zsh_async
- updated the code to run in the new environment

TODO:
There is still an issue with the BATTERY segment background color
being reset to %B{black} for some reason when using the background
colors array.

NOTE:
@dritter You might want to merge your version of this file first if you
want to keep merge history up to date.

onaforeignshore committed:
Code cleanup - see notes below

- moved local declarations out of if statements
- changed local offset to declare -i offset
  - This allows us to have an integer instead of a float for the array
  indices.
- combined p9kPreparePrompt calls into a single line

onaforeignshore committed:
ZSH-ASYNC update - see notes below

- updated rebuild_vi_mode
  - added current_state code
  - added VI_state_MODE_STRING
  - changed call to prompt_vi_mode to p9kAsyncCallback - this "cheat"
  allows us to regenerate the segment in real-time instead of an
  asynchronous call.

onaforeignshore committed:
Fix for dir segment in zsh-async mode

generator/zsh-async.p9k
- Passed the $PWD in the prompt_{$segment} call

segments/dir.p9k
- check if we are using the async-generator and that the extra parameter
has a value, then ch into that folder before doing anything else.

onaforeignshore committed:
See notes below

- changed the order of the test in back|foregroundColor
- It is more likely that a color will be specified than an empty string
will occur, so this speeds up the code by a fraction of a second, but it
is called multiple times...

onaforeignshore committed:
See notes below

- removed the local from defined
- since this function is called multiple times, it is faster not to
create a local every time, but to test $1 directly.

onaforeignshore committed:
Replaced printIcon with p9k_icons array

onaforeignshore committed:
Replaced printIcon with p9k_icons array

onaforeignshore committed:
Replaced printIcon with p9k_icons array

onaforeignshore committed:
See notes below

- Replaced printIcon with p9k_icons array
- Added workaround for zsh-async (hidden fourth paramater to change
working directory for sub shell)

onaforeignshore committed:
Replaced HOME_ICON with P9K_DIR_HOME_ICON

onaforeignshore committed:
Changes to support overide BG and FG - See notes

generator/default.p9k -
- p9kPrepareSegment
  - added parameters for background and foreground overide
  - changed all local variables to parameter variables, except
STATEFUL_NAME
  - removed CONDITION and test directly against shouldSegmentBePrinted
  - set overide background and/or foreground if required

generator/zsh-async.p9k
- p9kPrepareSegment
  - added parameters for background and foreground overide
  - changed all local variables to parameter variables, except
  STATEFUL_NAME
  - removed CONDITION and test directly against shouldSegmentBePrinted
  - add overide background and foreground to result string
- p9kAsyncCallback
  - set overide background and/or foreground if required

segments/battery.p9k
= added locals for overideIcon and overideBg
- added them as parameters to p9kPrepareSegment

onaforeignshore committed:
Merge branch 'next' into 070-registerSegment

onaforeignshore committed:
See notes below

Changed local declaration of P9K_RIGHT_PROMPT_ELEMENTS from

local P9K_RIGHT_PROMPT_ELEMENTS=()

to

local -a P9K_RIGHT_PROMPT_ELEMENTS; P9K_RIGHT_PROMPT_ELEMENTS=()

This is done to support ZSH 5.0.2, which cannot do one line array declations

onaforeignshore committed:
Updated testNewlineOnRpromptCanBeDisabled()

- Added a missing space

onaforeignshore committed:
Updated API files

onaforeignshore committed:
Updated laravel_version

- changed Java to Laravel in comments
- generated API file

onaforeignshore committed:
Updated authors in all files and API

onaforeignshore committed:
Updated truncate_with_folder_marker

- We need to test against an absolute path for this to work correctly

onaforeignshore commented:
zsh-async still has two major issues with segments not working that need to be fixed before it can be released:

- command_execution_time
- status

onaforeignshore committed:
Updated calls with always show overrides

- CONDITION is set to "[[ true ]]" to ensure that the segment displays
correctly

onaforeignshore committed:
Fixed P9K_HIDE_BRANCH_ICON=true issue

onaforeignshore committed:
See notes below

segments/*
- Updated codepoints to use "\uXXX" for standard unicode and "\U1XXXX"
for extended unicode
- Added missing codepoints

segments/user.p9k
- added new state "SUDOER" for users that have an /etc/sudoers entry
that specifies "NOPASSWD"
- /etc/sudoers etries that don't have "NOPASSWD" specified are treated
as "SUDO" state.

onaforeignshore committed:
Added test for status segment

onaforeignshore committed:
Updated changelog to include 0.6.5

onaforeignshore committed:
Move declaration of matching into strategy

onaforeignshore committed:
Merge branch 'next' into 070-registerSegment

onaforeignshore committed:
Updated array declaration

onaforeignshore committed:
Fix for truncate_unique

onaforeignshore committed:
Updated laravel_version

onaforeignshore committed:
see notes below

Updated prompt_p9k_teardown
- renamed hook functions (p9k_\* to p9k\*)
- added function to unhook from p9kChPwd

Fixed issue with RETVAL(S)

onaforeignshore committed:
Replaced ~ with $HOME

onaforeignshore commented:
Once the code separation is completed, I will create multiple PRs to implement all the changes here.

onaforeignshore committed:
Updated laravel_version test spec

onaforeignshore committed:
Applied #893

onaforeignshore committed:
Changed @source in comments

onaforeignshore committed:
Changed @source in comments

onaforeignshore committed:
Autoload support - see notes

- moved some functions out of functions/utilities.zsh
  - segmentShouldBeJoined
  - segmentShouldBePrinted
  - truncatePath
  - upsearch
- created folder functions/autoload for these functions
- added support to convert an INI file into shell variables
  -readIniFile

updated test/functions/utilities.spec

merged #893

onaforeignshore committed:
Lots of changes - Last commit

onaforeignshore committed:
Updated mistake in updateEnvironmentVars()

onaforeignshore committed:
Changed color reference table

onaforeignshore committed:
Simplified test (== replaced with =)

onaforeignshore committed:
TaskWarrior segment added

onaforeignshore committed:
TaskWarrior added

onaforeignshore committed:
Removed "none" for getColorCode()

onaforeignshore committed:
See notes below

- improved updateEnvironmentVars
- fixed "informations" to "information"

onaforeignshore committed:
Changed NerdFont icon

onaforeignshore committed:
See notes below

- enabled range 8-15
- updated color table to use 3-digit codes (to keep testing working)

onaforeignshore committed:
Updated background segment

onaforeignshore committed:
New tests from @dritter

onaforeignshore committed:
Changes to allow new test suite to work

onaforeignshore committed:
See notes below

- Integrated & updated tests
- Fixed errors in segments

onaforeignshore committed:
Updated context & user to be the same

onaforeignshore committed:
Updated test - vcs-git.spec

onaforeignshore committed:
Updated suite.spec

- changed test for failure

onaforeignshore committed:
Added style guides for coding & wiki

onaforeignshore committed:
Updates to segments

onaforeignshore committed:
Updated logic and test

onaforeignshore committed:
Updated test

onaforeignshore committed:
Fixed test (extra space)

onaforeignshore committed:
renamed everything to match coding style

onaforeignshore committed:
Renaming variable error

onaforeignshore committed:
Updated utilities test with new name

onaforeignshore committed:
updated function name - p9k::defined

onaforeignshore committed:
Updated travis to include summary at end

onaforeignshore committed:
Added + to pass and - to fail message

onaforeignshore committed:
Renamed function - p9k::defined

onaforeignshore committed:
See notes below

Updated test suite to check for "-" in a filename
- this is used as a delimiter for an OS specific test
- this was done because the ip tests for Linux fail on OSX

onaforeignshore committed:
Made ip-linux.spec executable

onaforeignshore committed:
Added version number readonly __P9K_VERSION

This will make it easier to support as users can report this with
tickets.

onaforeignshore committed:
Updated test to support dynamic 'tmp' unique paths

onaforeignshore committed:
Updated tests to use latest shunit2

onaforeignshore committed:
Changed xcode version for Travis

onaforeignshore committed:
Updated colors - see notes

- Fixed naming convention
- Moved color translation table to global

onaforeignshore committed:
See notes below

- fixed indentation
- removed double calls to p9k::registerSegment

onaforeignshore committed:
Fixes from #923

onaforeignshore committed:
Renamed p9k_data to __P9K_DATA

onaforeignshore committed:
Renamed p9kDirectory to __P9K_DIRECTORY

onaforeignshore committed:
Renamed p9k_data to __P9K_DATA

onaforeignshore committed:
Renamed globals

- p9k_data to __P9K_DATA
- p9kDirectory to __P9K_DIRECTORY

onaforeignshore committed:
Renamed p9kDirectory to __P9K_DIRECTORY

onaforeignshore committed:
Renamed p9kDirectory to __P9K_DIRECTORY

onaforeignshore committed:
Code cleanup

onaforeignshore committed:
Various - see notes below

- renamed VISUAL_IDENTIFIER to SEGMENT_ICON
- merged functions/vcs.zsh into segments/vcs.p9k - single file for
segment and additional functions only loaded if needed
- moved __p9k_update_var_name from utilities.zsh to autoload
- fixed tests
- various other fixes

onaforeignshore committed:
Updated api files

onaforeignshore committed:
Added zsh-async as submodule

onaforeignshore committed:
Replaced $(pwd) with $PWD
